ecommerce/products/urls.py

```diff
@@ -4,6 +4,6 @@
 
 urlpatterns = [
     path('', views.HomePageView.as_view(), name='home'),
-    path('<product_type>', views.ProductByTypeView.as_view()),
+    path('<product_type>', views.ProductByTypeView.as_view(), name='product-by-type'),
```

Comment:
Ah yes, I remember what I did to fix the 404 issue. It has something do do with the order of url patterns.
Here's what you url patterns are doing right now:
```
    # If the url is '' exactly, pass request to `HomePageView`, otherwise try next url pattern
    path('', views.HomePageView.as_view(), name='home'),
    
    # If the url is any value, pass request to `ProductByTypeView`, otherwise try next url pattern
    path('<product_type>', views.ProductByTypeView.as_view(), name='product-by-type'),
    
    # If the url is an integer, pass request to `ProductDetailView `, otherwise raise 404 (no more patterns to check)
    path('<product_type>', views.ProductByTypeView.as_view(), name='product-by-type'),
```

Reply:
So the second pattern is catching everything. If you swap the order of the second one and the third one it will start working as expected. Try this and see if it works. Also try make sure you understand _why_ this is happening